components/utils.py

```diff
@@ -520,3 +606,282 @@ def untar(source, destination='/tmp', strip=1):
+        return dest_file_path
+
+    def get_upgrade_resources_json(self, service_name):
+        return self._get_resources_json(service_name, True)
+
+    def get_install_resources_json(self, service_name):
+        return self._get_resources_json(service_name, False)
+
+    def _get_resources_json(self, service_name, upgrade):
+        resources_json = self._get_resource_file_path(service_name,
+                                                      self.RESOURCES_JSON_FILE,
+                                                      upgrade)
+        if os.path.isfile(resources_json):
+            with open(resources_json) as f:
+                return json.load(f)
+        return {}
+
+    def _set_resources_json(self, resources_dict, service_name, upgrade):
+        resources_json = self._get_resource_file_path(service_name,
+                                                      self.RESOURCES_JSON_FILE,
+                                                      upgrade)
+        write_file(resources_dict, resources_json)
+
+    def archive_resources(self, service_name):
+        rollback_dir = os.path.join(self.RESOURCES_ROLLBACK_PATH, service_name)
+        mkdir(rollback_dir)
+
+        service_archive_path = os.path.join(self.RESOURCES_PATH, service_name)
+        # nodes will not always use blueprint resources i.e. java/python
+        if os.path.isdir(service_archive_path) and self.is_upgrade:
+            ctx.logger.info('archiving node {node} properties'
+                            .format(node=service_name))
+            resource_files = os.listdir(service_archive_path)
+            for filename in resource_files:
+                if not filename.startswith(self.UPGRADE_RESOURCE_PREFIX):
+                    move(os.path.join(service_archive_path, filename),
+                         rollback_dir)
+
+            for filename in resource_files:
+                if filename.startswith(self.UPGRADE_RESOURCE_PREFIX):
+                    move(os.path.join(service_archive_path, filename),
+                         os.path.join(service_archive_path, filename[8:]))
+
+
+def is_upgrade():
+    '''
+    Returns true if manager is in upgrade mode. This function will assume
+    manager is in upgrade state according to the maintenance mode file.
+    :return: True if in upgrade state, false otherwise.
+    '''
+    status_file_path = '/opt/manager/status.txt'
+    if os.path.isfile(status_file_path):
+        ctx.logger.info('loading upgrade status file: {0}'
+                        .format(status_file_path))
+        with open(status_file_path) as f:
+            status_dict = json.load(f)
+        return status_dict['status'] == 'activated'
+    else:
+        return False
+
+
+def is_rollback():
+    '''
+    Returns true if manager is in rollback state.
+    :return: True if in rollback state, false otherwise.
+    '''
+    return False
+
+
+def start_service_and_archive_properties(service_name,
```

Comment:
the name of this function is misleading. using it I would think that it does exactly what it says. thing is.. if I use it outside of an upgrade context, it would just start a service. If I use it inside an upgrade context, it doesn't just archive properties and restart.. it restarts a service and archives both properties and resources. I suggest we make the name more generic.